service/src/main/resources/api/service_openapi.yaml

```diff
@@ -2386,6 +2456,75 @@ components:
       items:
         $ref: "#/components/schemas/ExportModel"
 
+    ResourceType:
+      type: string
+      enum: [ "STUDY", "COHORT", "REVIEW", "CONCEPT_SET" ]
+
+    Resource:
+      type: object
+      properties:
+        type:
+          $ref: "#/components/schemas/ResourceType"
+        studyId:
+          $ref: "#/components/schemas/StudyIdV2"
+          nullable: false
+        cohortId:
+          $ref: "#/components/schemas/CohortIdV2"
+          nullable: true
+        cohortRevisionId:
+          $ref: "#/components/schemas/CohortRevisionIdV2"
+          nullable: true
+        reviewId:
+          $ref: "#/components/schemas/ReviewIdV2"
+          nullable: true
+        conceptSetId:
+          $ref: "#/components/schemas/ConceptSetIdV2"
+          nullable: true
+      required:
+        - studyId
+
+    ActivityType:
+      type: string
+      enum: [ "CREATE_STUDY", "DELETE_STUDY", "CREATE_COHORT", "DELETE_COHORT", "EXPORT_COHORT", "CREATE_REVIEW", "DELETE_REVIEW", "CREATE_CONCEPT_SET", "DELETE_CONCEPT_SET" ]
```

Comment:
Thoughts on editing cohorts/"data features sets"? Is it enough to capture at review and export time?

Reply:
I asked VUMC before and they said just cohorts at create/delete/review/export. So I guess I could reduce this list of events down to just those 4, but I thought it would be relatively straightforward to add the others once I have all the tables. If it turns into more work than I'm expecting, I'll skip the other ones.

RE cohort edit: This sounds useful to me, but potentially hard to implement because isn't the UI auto-saving every time someone adds a new criteria? Might make the audit log cluttered, unless we had some timer logic (e.g. freeze the cohort revision at most every 1 day).

Reply:
Per offline discussion, we agreed that supporting any additional audit events can come post-prod, without any major changes.